src/EditorFeatures/CSharpTest/UsePatternMatching/CSharpAsAndNullCheckTests.cs

```diff
@@ -315,6 +315,66 @@ void M()
 }");
         }
 
+        [WorkItem(33345, "https://github.com/dotnet/roslyn/issues/33345")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestRemoveNewLines()
+        {
+            await TestInRegularAndScriptAsync(
+                @"class C
+{
+    void M()
+    {
+        [|var|] x = o as string;
+
+        //suffix comment
+        if (x != null)
+        {
+        }
+    }
+}",
+                @"class C
+{
+    void M()
+    {
+        //suffix comment
+        if (o is string x)
+        {
+        }
+    }
+}");
+        }
+
+        [WorkItem(33345, "https://github.com/dotnet/roslyn/issues/33345")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task TestRemoveNewLines2()
+        {
+            await TestInRegularAndScriptAsync(
+                @"class C
+{
+    void M()
+    {
+        int a = 0;
+        [|var|] x = o as string;
+
+        //suffix comment
+        if (x != null)
+        {
+        }
+    }
+}",
+                @"class C
+{
+    void M()
+    {
+        int a = 0;
+        //suffix comment
```

Comment:
this does not seem like desirable behavior.  i think it's fine to remove the leading blanks if the 'if' becomes the first statement in the block.  But if there was a separation originally, i think we should keep that around.

Reply:
@CyrusNajmabadi I agree. I'll see if I can find a way to make the editor smarter like you suggested in https://github.com/dotnet/roslyn/issues/33345#issuecomment-464652548

Reply:
@CyrusNajmabadi The thing is, we already have a thing to clean up whitespace: the static methods in `Formatter`. But they need a `Workspace` as argument. So what to do here? Give the code fix provider access to a `Workspace` (through the constructor for example) and then after doing all the fixes simply call one of `Formatter`'s methods? Somehow that doesn't feel like the way to go.

Reply:
1. there are several formatter methods that don't take a workspace.  
2. however, this won't really help.  Code Actions *already* send their code through the formatter by default.  however, the formatter isn't really aggressive about making these sorts of changes.  Instead, the onus is on the code-action to examine the code for common scenarios and make appropriate changes for at least the 80-90% cases.

Reply:
I see! Thanks, that helps.